internal/biz/model/resource.go

```diff
@@ -125,6 +129,10 @@ func (r *Resource) Update(
 		return fmt.Errorf("failed to create updated ResourceReportEvent: %w", err)
 	}
 
+	// Preserve the original created_at and set updated_at to current time
+	existingCreatedAt := r.resourceReportEvents[0].createdAt
```

Comment:
We should check that this slice is non empty, similar to how it's done in `GetTimestamps()`